internal/dim-2d/dim-2d.go

```diff
@@ -105,6 +107,7 @@ func GetAndSaveData(ctx context.Context, dataDirectory string, lidar lidar.Lidar
 	ctx, span := trace.StartSpan(ctx, "viamcartographer::internal::dim2d::GetAndSaveData")
 	defer span.End()
 
+	ctx, md := utils.ContextWithMetadata(ctx)
```

Comment:
Based on the results of the discussion regarding my open question, update the comment here to more correctly represent that the deciding factor is the parsing/extraction and not the presence of the timestamp.

Example:
```suggestion
	// If the timestamps correlated with the point cloud and provided by the server can be extracted use them when writing the data to disk, otherwise use the current time.
```